### PR TITLE
✨Source Google Ads: Add possibility to sync all connected accounts

### PR DESCRIPTION
**Pull request by tolik0**
## What

- **Implemented Account Synchronization**: This update introduces the capability to sync all connected accounts, resolving the issue: https://github.com/airbytehq/oncall/issues/3741.

- **Removed `login_customer_id` Property**: To simplify configuration, the `login_customer_id` property has been removed from the spec. Previously, this property was essential for accessing client accounts under a manager account. The Google Ads connector automatically retrieves this information while requesting connected customers.

- **Introduced `customer_status_filter` Property**: A new property, `customer_status_filter`, has been added. This feature enables users to filter out inactive accounts to speed up the sync.

## How

The `customer_id` property has been revised to be optional. In this new setup, the connector initially requests data from all accessible accounts and then proceeds to iterate over the child customers associated with these accounts. This enhancement significantly streamlines the configuration process, allowing for the aggregation of multiple customers into a single connection. Moreover, this update ensures minimal configuration, requiring only authentication. This simplification greatly reduces setup complexity and enhances user convenience.

### CODE REVIEW

**Comment by alafanechere** [comment]
Hey @tolik0 . As we're under code freeze, would you mind re-creating your PR with a different branch name? 
Please follow the instructions [here](https://gist.github.com/alafanechere/260ec104be946c550bb81cd210632b9f)

**Comment by alafanechere** [comment]
@tolik0 could you please let me know if this is a bug fix or a new feature?
I know it's related to an OC issue, but I'm not sure if:
* this is an improvement to better handle the problem the user faced but there is a workaround for them to get their connection working
* this is a bug fix, the impacted customer can't set their connection otherwise

**Comment by tolik0** [comment]
@alafanechere, this update is essentially a new feature crucial for a specific user scenario. Currently, our Google Ads connector syncs data exclusively from preselected accounts. The user who requested this feature manages over 100 connected customers, making it challenging to manually identify and input all the necessary IDs. The tree-like structure of the customer accounts further compounds this complexity. Therefore, this feature automates the process, enabling efficient synchronization across all connected customer accounts.

**Review by alafanechere — COMMENTED** [review]
As this is quite a substantial logic change, addressing a single customer issue, I feel it'd be safer to:
* Not merge this during the code freeze
* Publish a pre-release version
* Coordinate with the support team to make the customer create a new connection using this specific version and confirm their problem is addressed.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py]: Fixed to `default`

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py]: Fixed

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/models.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/models.py]: Fixed

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py:62]: Google Ads doesn't use permissions specific to individual endpoints. We call this endpoint during the check, and if there are any access-related issues, an error will be raised accordingly.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:150]: The previous version raised an error when accessing data for inaccessible customer IDs. This was because the connector could not retrieve data from disabled accounts. In the new version, I've updated the behaviour to enhance robustness. Now, instead of failing, the connector logs a warning when it encounters inaccessible accounts. This change aims to avoid errors due to disabled accounts.

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:146]: Fetching all connected accounts involves just a single request. Moreover, this endpoint is the only way to verify access to a customer, especially when there are no custom queries involved.

File: airbyte-integrations/connectors/source-google-ads/unit_tests/test_source.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/unit_tests/test_source.py:1]: Added test for `get_customers`

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/models.py
Comment (tolik0) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/models.py]: Fixed

**Review by alafanechere — APPROVED** [review]
Thanks for your changes and replies. LGTM.

**Comment by nazariyme** [comment]
Hey,
these changes should be reverted as it causes failure for Google Ads source if there's customer ID with not finished setup
All our connections failed today
Screenshot_1
Screenshot_2

**Comment by tolik0** [comment]
Hi @nazariyme. Thank you for bringing this to our attention. I have created a [PR](https://github.com/airbytehq/airbyte/pull/34200) to address the issue and will let you know as soon as it's merged. I'm very sorry for any inconvenience this has caused.

**Comment by nazariyme** [comment]
I fixed it by removing that pending customer ID as it was not used by anyone. All works fine for us now but it worth fixing in the source itself. @tolik0 thanks for a quick response and the fix you prepared

**Comment by bachik-24** [comment]
Hi @tolik0 
After this release, I corrected the name of the variable in the source json from custom_queries to custom_queries_array. Now the connection does not take a single line, but it runs successfully. With what it can be connected?

**Comment by tolik0** [comment]
@bachik-24 Hi, what do you mean by "Connection does not take a single line"?

**Comment by bachik-24** [comment]
@tolik0  
I mean that the source is configured correctly but does not return a single row (0 bytes, 0 records). On the previous version with the same parameters everything works. What could go wrong?

**Comment by tolik0** [comment]
@bachik-24  Can you please share your `custom_queries_array` parameter?

**Comment by bachik-24** [comment]
@tolik0 
```
                    {
                        "query": "SELECT segments.device, customer.id, metrics.cost_micros, campaign.id, campaign.name, metrics.clicks, metrics.conversions, metrics.impressions, geographic_view.country_criterion_id, geographic_view.location_type, geographic_view.resource_name FROM geographic_view",
                        "table_name": "google_acquisition_report_5197072984"
                    }
```

**Comment by tolik0** [comment]
@bachik-24 It looks like you should add the `segments.date` parameter to your query; previously, it was added automatically. Here is the PR with these changes: https://github.com/airbytehq/airbyte/pull/33603

**Comment by bachik-24** [comment]
@tolik0  This helped, thank you very much! I think this should be indicated separately in the commit that now this parameter is worth specifying.

**Comment by tolik0** [comment]
@bachik-24 This change is described in the PR description, but actually, you don't need to modify your config manually, as it is updated using config migrations when you run connector with the old config: https://github.com/airbytehq/airbyte/blob/66ce98fdc4eacd6a3d69210630e3b3b9e0762ba9/airbyte-integrations/connectors/source-google-ads/source_google_ads/config_migrations.py

**Comment by bachik-24** [comment]
@tolik0  thank you for your work!

**Pull request merged**
